src/main/java/games/strategy/triplea/delegate/Matches.java

```diff
@@ -1811,18 +1834,23 @@ private Matches() {}
     });
   }
 
-  public static final Match<Unit> UnitRequiresUnitsOnCreation = Match.of(unit -> {
-    final UnitAttachment ua = UnitAttachment.get(unit.getType());
-    if (ua == null) {
-      return false;
-    }
-    return (ua.getRequiresUnits() != null && ua.getRequiresUnits().size() > 0);
-  });
+  /**
+   * Returns a match indicating the specified unit requires at least one type of unit upon creation.
+   */
+  public static Match<Unit> unitRequiresUnitsOnCreation() {
+    return Match.of(unit -> {
+      final UnitAttachment ua = UnitAttachment.get(unit.getType());
+      if (ua == null) {
+        return false;
+      }
+      return ua.getRequiresUnits() != null && ua.getRequiresUnits().size() > 0;
```

Comment:
The if check can be merged with this return statement: `return ua != null && ua.getRequiresUnits() != null && ua.getRequiresUnits().size() > 0;`